cache/remotecache/s3/s3.go

```diff
@@ -209,14 +209,22 @@ func (e *exporter) Finalize(ctx context.Context) (map[string]string, error) {
 			}
 		} else {
 			layerDone := progress.OneOff(ctx, fmt.Sprintf("writing layer %s", l.Blob))
-			dt, err := content.ReadBlob(ctx, dgstPair.Provider, dgstPair.Descriptor)
-			if err != nil {
-				return nil, layerDone(err)
-			}
-			if err := e.s3Client.saveMutable(ctx, key, dt); err != nil {
-				return nil, layerDone(errors.Wrap(err, "error writing layer blob"))
+			if int64(len(dgstPair.Descriptor.Data)) == dgstPair.Descriptor.Size && digest.FromBytes(dgstPair.Descriptor.Data) == dgstPair.Descriptor.Digest {
```

Comment:
I think these digests can never have inline data. If we want to handle this case for safety then we should write a helper function (in `contentutil` maybe, or maybe containerd wants it) `ReaderAt(ctx, Provider, desc)` that also handles the inline data.

Reply:
Good point, moving this there !

Reply:
Gaining a little bit of traction over there with a very good review simplifying the code, https://github.com/containerd/containerd/pull/9657 Will update this PR if///when...

Reply:
Okay, https://github.com/containerd/containerd/pull/9657 was recently merged. Rebasing & refreshing this.

Reply:
Oh, wait they are releasing a v2 it seems 🤔

Reply:
Yeah, don't update to v2 as part of this PR. You can leave a comment that this needs to be updated after upgrade to containerd v2 or you can just ignore this case as inline descriptor data should be impossible atm.

Reply:
Gotcha, I pushed a change without the inline data check and a TODO, lmk if anything